explainaboard/feature.py

```diff
@@ -278,350 +18,123 @@ class BucketInfo:
-
-    def int2str_vector(self, values: Iterable[int]) -> list[str]:
-        """Convert a list of class IDs into corresponding names."""
-        return [self.int2str_scalar(v) for v in values]
-
-    @overload
-    def int2str(self, values: int) -> str:
-        ...
-
-    @overload
-    def int2str(self, values: Iterable[int]) -> list[str]:
-        ...
-
-    def int2str(self, values):
-        """Convert class IDs into corresponding names: generic version."""
-        if isinstance(values, int):
-            return self.int2str_scalar(values)
-        else:
-            return self.int2str_vector(values)
-
-    def encode_example(self, example_data):
-        if self.num_classes is None:
-            raise ValueError(
-                "Trying to use ClassLabel feature with undefined number of class. "
-                "Please set ClassLabel.names or num_classes."
-            )
+    require_training_set: bool = False
+    id: Optional[str] = None
 
-        # If a string is given, convert to associated integer
-        if isinstance(example_data, str):
-            example_data = self.str2int(example_data)
+    # @classmethod
+    # def from_dict(cls, data_dict: dict) -> FeatureType:
+    #
+    #     field_names = set(f.name for f in dataclasses.fields(cls))
+    #     return cls(**{k: v for k, v in data_dict.items() if k in field_names})
 
-        # Allowing -1 to mean no label.
-        if not -1 <= example_data < self.num_classes:
-            raise ValueError(
-                "Class label %d greater than configured num_classes %d"
-                % (example_data, self.num_classes)
-            )
-        return example_data
+    @classmethod
+    def from_dict(cls, obj: dict) -> FeatureType:
+        if not is_dataclass_dict(obj):
+            raise TypeError("from_dict() should be called on dict with _type")
+        return _fromdict_inner(obj)
 
-    @staticmethod
-    def _load_names_from_file(names_filepath):
-        with open(names_filepath, "r", encoding="utf-8") as f:
-            return [
-                name.strip() for name in f.read().split("\n") if name.strip()
-            ]  # Filter empty names
+    def __post_init__(self):
+        self._type: str = self.__class__.__name__
 
 
 @dataclass
-class Sequence:
-    """Construct a list of feature from a single type or a dict of types.
-    Mostly here for compatiblity with tfds.
-    """
+class Sequence(FeatureType):
+    feature: FeatureType = field(default_factory=FeatureType)
 
-    feature: Any
-    length: int = -1
-    id: Optional[str] = None
-    is_bucket: bool = False
-    require_training_set: bool = False
-    # Automatically constructed
-    dtype: ClassVar[str] = "list"
-    pa_type: ClassVar[Any] = None
-    _type: str = field(default="Sequence", init=False, repr=False)
+    def __post_init__(self):
+        super().__post_init__()
+        self.dtype = "list"
 
 
 @dataclass
-class Set:
-    feature: dict
+class Set(FeatureType):
```

Comment:
I know this is something that existed before, but it seems a little bit weird for me to have this called `Set` when the underlying data type is `dict`. Maybe changing it to be consistent would be better? Similarly, we could potentially change `Sequence` to `List`?